crates/red_knot_python_semantic/resources/mdtest/type_of/any.md

```diff
@@ -0,0 +1,46 @@
+# type[Any]
+
+## Simple
+
+```py
+def f(x: type[Any]):
+    reveal_type(x)  # revealed: type[Any]
+    # TODO: could be `<object.__repr__ type> & Any`
+    reveal_type(x.__repr__)  # revealed: Any
+
+class A: ...
+
+x: type[Any] = object
+x: type[Any] = type
+x: type[Any] = A
+```
```

Comment:
It makes me slightly nervous that all the assignment tests in this file are "no error" assignments; we could make all these tests pass if `type[...]` were always just `Any`! (Not the `reveal_type` ones, but the assignment ones.) We could add something like this to show that we do actually discriminate?
```suggestion
x: type[Any] = A()  # error: [invalid-assignment]
```
And maybe similar in below sections, too?

Reply:
Done